packages/builtin/string.pony

```diff
@@ -1352,3 +1335,50 @@ class StringRunes is Iterator[U32]
     (let rune, let len) = _string.utf32(_i.isize())
     _i = _i + len.usize()
     rune
+
+primitive _UTF32Encoder
+  fun encode(value: U32): (USize, U8, U8, U8, U8) =>
+    """
+    Encode the code point into UTF-8. It returns a tuple with the size of the encoded data and then the data
+    """
+    if value < 0x80 then
+      (1, value.u8(), 0, 0, 0)
+    elseif value < 0x800 then
+      (
+        2,
+        ((value >> 6) or 0xC0).u8(),
+        ((value and 0x3F) or 0x80).u8(),
+        0,
+        0
+      )
+    elseif value < 0xD800 then
+      (
+        3,
+        ((value >> 12) or 0xE0).u8(),
+        (((value >> 6) and 0x3F) or 0x80).u8(),
+        ((value and 0x3F) or 0x80).u8(),
+        0
+      )
+    elseif value < 0xE000 then
+      // UTF-16 surrogate pairs are not allowed.
+      (3, 0xEF, 0xBF, 0xBD, 0)
+    elseif value < 0x10000 then
+      (
+        3,
+        ((value >> 12) or 0xE0).u8(),
+        (((value >> 6) and 0x3F) or 0x80).u8(),
+        ((value and 0x3F) or 0x80).u8(),
+        0
+      )
+    elseif value < 0x110000 then
+      (
+        4,
+        ((value >> 18) or 0xF0).u8(),
+        (((value >> 12) and 0x3F) or 0x80).u8(),
+        (((value >> 6) and 0x3F) or 0x80).u8(),
+        ((value and 0x3F) or 0x80).u8()
+      )
```

Comment:
I would probably tend to use a code style that looks like yours here in my own code, but I don't necessarily think that this fits well with the code style for the pony standard library.  It's not particularly clear cut as I don't think we have any explicit rules for this situation, but I think the other rules and de-facto standards for multiline argument lists would apply, and would imply a formatting that is something like:

```pony
(4,
  ((value >> 18) or 0xF0).u8(),
  (((value >> 12) and 0x3F) or 0x80).u8(),
  (((value >> 6) and 0x3F) or 0x80).u8(),
  ((value and 0x3F) or 0x80).u8())
```

@ponylang/core @ponylang/committer any thoughts on this?